src/main/java/com/revature/services/PersonBatchLogicImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.revature.services;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.revature.domain.Person;
+import com.revature.domain.PersonBatchJoin;
+import com.revature.repositories.PersonBatchRepository;
+
+@Service
+@Transactional(readOnly=false, isolation=Isolation.READ_COMMITTED)
+public class PersonBatchLogicImpl implements PersonBatchLogic{
+
+	@Autowired
+	PersonBatchRepository pBJ;
+	
+	@Override
+	public Page<Person> getAllBatchMembers(Pageable pageable, Integer batchId) {
+		List<PersonBatchJoin> list = pBJ.findAllByBatchId(batchId);
+		List<Person> personList = new ArrayList<>();
+		for(PersonBatchJoin temp:list){
+			personList.add(temp.getPerson());
+		}
+		Page<Person> personPage = new PageImpl<Person>(personList);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS2293)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "personPage". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS1488)